In [9]:
import pandas as pd
import matplotlib.pyplot as plt
df = pd.read_csv("clean_data.csv")
print(df)

        Unnamed: 0         ID Customer_ID      Month             Name   Age  \
0                0   \n0x160a   CUS_0xd40  September    Aaron Maashoh  23.0   
1                1   \n0x160b   CUS_0xd40    October    Aaron Maashoh  24.0   
2                2   \n0x160c   CUS_0xd40   November    Aaron Maashoh  24.0   
3                3   \n0x160d   CUS_0xd40   December    Aaron Maashoh   NaN   
4                4   \n0x1616  CUS_0x21b1  September  Rick Rothackerj  28.0   
...            ...        ...         ...        ...              ...   ...   
149996      149996  \n0x25fea  CUS_0x942c        May            Nicks  25.0   
149997      149997  \n0x25feb  CUS_0x942c       June            Nicks  25.0   
149998      149998  \n0x25fec  CUS_0x942c       July            Nicks  25.0   
149999      149999  \n0x25fed  CUS_0x942c     August            Nicks  25.0   
150000      150000         \n         NaN        NaN              NaN   NaN   

       Occupation  Annual_Income  Monthly_Inhand_Sa

C:\Users\Wei Xun\AppData\Local\Temp\ipykernel_14292\3309926209.py:3: DtypeWarning: Columns (27) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("clean_data.csv")


num_of_Delayed_Payment

In [10]:
df['Num_of_Delayed_Payment'].describe()

count    139500.000000
mean         30.934301
std         224.530918
min           0.000000
25%           9.000000
50%          14.000000
75%          18.000000
max        4399.000000
Name: Num_of_Delayed_Payment, dtype: float64

In [11]:
upper = df['Num_of_Delayed_Payment'].mean()+(2*df['Num_of_Delayed_Payment'].std())

for x in range(len(df['Num_of_Delayed_Payment'])):
    if df['Num_of_Delayed_Payment'][x] > upper:
        df['Num_of_Delayed_Payment'][x] = int(upper)

df['Num_of_Delayed_Payment'].fillna(df['Num_of_Delayed_Payment'].mode()[0], inplace=True)
df['Num_of_Delayed_Payment'].describe()

C:\Users\Wei Xun\AppData\Local\Temp\ipykernel_14292\800859967.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Num_of_Delayed_Payment'][x] = int(upper)


count    150001.000000
mean         17.080199
std          39.402941
min           0.000000
25%           9.000000
50%          15.000000
75%          19.000000
max         479.000000
Name: Num_of_Delayed_Payment, dtype: float64

I'm using 2 SD (95% of data), and using the upper boundary to replace the ones out of bounds, and the max (479) is the number of delayed payments over a few years so its understandable too

Num_Credit_Inquiries (when a creditor checks your credit report, could hurt your credit score if its a hard inquiry, where you apply for a credit card, mortgage, auto loan, or credit increase, and it won't hurt your credit score if its a soft inquiry, where you apply for insurance, or request your credit report)

In [12]:
df['Num_Credit_Inquiries'].describe()

count    147000.000000
mean         28.529014
std         194.456058
min           0.000000
25%           3.000000
50%           6.000000
75%           9.000000
max        2597.000000
Name: Num_Credit_Inquiries, dtype: float64

In [13]:
upper = df['Num_Credit_Inquiries'].mean()+(2*df['Num_Credit_Inquiries'].std())

for x in range(len(df['Num_Credit_Inquiries'])):
    if df['Num_Credit_Inquiries'][x] > upper:
        df['Num_Credit_Inquiries'][x] = int(upper)

df['Num_Credit_Inquiries'].fillna(df['Num_Credit_Inquiries'].mode()[0], inplace=True)
df['Num_Credit_Inquiries'].describe()

C:\Users\Wei Xun\AppData\Local\Temp\ipykernel_14292\1158527260.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Num_Credit_Inquiries'][x] = int(upper)


count    150001.000000
mean         12.578616
std          50.054023
min           0.000000
25%           3.000000
50%           6.000000
75%           9.000000
max         417.000000
Name: Num_Credit_Inquiries, dtype: float64

I'm using 2 SD (95% of data), and using the upper boundary to replace the ones out of bounds, which is kind of understandable when someone has made 417 inquiries over like a few years, and they could do this by viewing their own credit too

Outstanding_Debt (debt that is yet to be paid)

In [14]:
df['Outstanding_Debt'].describe()

count    150000.000000
mean       1426.220376
std        1155.127101
min           0.230000
25%         566.072500
50%        1166.155000
75%        1945.962500
max        4998.070000
Name: Outstanding_Debt, dtype: float64

I don't think this needs cleaning

Credit_Utilization_Ratio (the percentage of a borrower’s total available credit that is currently being used)

In [15]:
df['Credit_Utilization_Ratio'].describe()

count    150000.000000
mean         32.283309
std           5.113315
min          20.000000
25%          28.054731
50%          32.297058
75%          36.487954
max          50.000000
Name: Credit_Utilization_Ratio, dtype: float64

I don’t think this needs cleaning, but keep in mind its percentage and not 0-1 ratio